internal/logger/logger.go

```diff
@@ -103,7 +103,14 @@ func (l *Logger) AddBlockStream(
 	if err != nil {
 		return err
 	}
-	defer f.Close()
+
+	defer func() {
```

Comment:
Can we make this a helper function? I don't like the idea of copying the same code N times in the same file.